board/stm32h7/sound.h

```diff
@@ -31,13 +34,21 @@ void sound_tick(void) {
 static void DMA1_Stream0_IRQ_Handler(void) {
   __attribute__((section(".sram4"))) static uint16_t tx_buf[MIC_TX_BUF_SIZE];
 
-  DMA1->LIFCR |= 0x7D; // clear flags
+  DMA1->LIFCR |= 0x7DU; // clear flags
 
-  // process samples
-  uint8_t buf_idx = (((DMA1_Stream0->CR & DMA_SxCR_CT) >> DMA_SxCR_CT_Pos) == 1U) ? 0U : 1U;
-  for (uint16_t i=0U; i < MIC_RX_BUF_SIZE; i++) {
-    tx_buf[2U*i] = ((mic_rx_buf[buf_idx][i] >> 16U) & 0xFFFFU);
-    tx_buf[(2U*i)+1U] = tx_buf[2U*i];
+  if (mic_buffer_count < MIC_SKIP_BUFFERS) {
+    // Send silence during settling
+    mic_buffer_count++;
+    for (uint16_t i = 0U; i < MIC_TX_BUF_SIZE; i++) {
+      tx_buf[i] = 0U;
```

Comment:
Don't think `0U` is correct here? it's a signed signal inside of an unsigned buffer, so we probably want the actual signal "zero" point which should be roughly `(1U << 15)`

Reply:
Your logic feels correct at first and that's what I was thinking too, but tests showed that it's incorrect.
With 0U I am getting nice silence at 0 level, but with 1U << 15 it's clipping to negative value.
<img width="650" height="422" alt="Screenshot 2025-10-29 at 1 49 11 PM" src="https://github.com/user-attachments/assets/abd3db5e-5f70-4089-b676-145fddce12c9" />

Reply:
Not sure how, but okay